### ISSUE TITLE
[BUG] <World doesn't load, timed out waiting for world statistics>

### ISSUE THREAD

**Issue by NoMoreStars** [ldtteam/minecolonies#9375]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.1.153 BETA

### Structurize Version

1.0.631 BETA

### Related Mods and their Versions

The pack is All the Mods 9 0.1.0 and here is the modlist:
![image](https://github.com/ldtteam/minecolonies/assets/38673759/c3cd9171-7f24-4709-8819-cf7c91406f2b)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/8caa8020-6435-42d8-8eab-9cc44fddbef8)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/d807ac83-83a0-4f0a-a938-b04dad4823fc)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/8d71cb55-acae-45c1-87db-a91f2cf8b249)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/f66f35be-4a98-4ecf-b97a-34b9badf98cc)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/d1878c93-8925-43a7-a757-95bfe12f0ee6)
![image](https://github.com/ldtteam/minecolonies/assets/38673759/4cba105f-a1b9-42de-b273-9f65fe294531)

### Current Behavior

When loading a world with an active Minecolony, the loading process becomes stuck waiting for world statistics, specifically it seems stuck on something regarding colonist pathfinding, see log for more details. I don't know what's causing it but it, I suspected it was just a corrupted chunk as I saw someone mentioning that on the Discord. But I personally think it could be related to the fact that this consistently happens whenever my Minecolony reaches 12 citizens in size. I reverted to a backup where I had only 11, it worked fine. But once the colony reached 12 again I got the same issue.

### Expected Behavior

World should load without issue

### Reproduction Steps

1.Create Colony
2.Create save backup at 11 citizens
3.Reach 12 citizens
4.Save and quit world
5.Reload save

### Logs

https://mclo.gs/IqWI2Tm

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
Can you try without modernfix?

**Comment by NoMoreStars** [comment]
Got modernfix disabled, same thing is happening game-side but looks like I'm getting slightly different results gamelog-side and I can't tell if Minecolonies or not involved since Silent Gear is loading before the server times out https://mclo.gs/EaLCDIo

**Comment by Raycoms** [comment]
We got a pending fix on structurize atm

**Comment by NoMoreStars** [comment]
Could I get a Structurize build with the pending fix on my end? I'd build it myself but gradle and JDE don't exactly play nice with my Windows installation for some reason

**Comment by NoMoreStars** [comment]
Updated to the latest version of Structurize, still getting a similar issue with the latest log here https://mclo.gs/GgemrOd
I did see ignore if single player but figured I'd try anyway

**Comment by Raycoms** [comment]
Can you send us the world?

**Comment by NoMoreStars** [comment]
Sure thing! There's the world in the state that lead to the most recent log submitted above as well as a backup from a week ago that should hopefully be before the bug started, but it might not be as the Simple Backups mod overwrites backups after a while so it may have gotten overwritten. https://mega.nz/folder/HihHHYrL#nYCYUplyHKBKsp2sYBRlCQ

**Comment by NoMoreStars** [comment]
Seems like the bug was fixed with the latest version of Minecolonies, I got the world back up and running again until I updated from ATM9 0.1.1 -> ATM9 0.1.4 and now I'm getting the timed out error again, here's the [log](https://mclo.gs/NuDLSoh) and the most recent [world download](https://mega.nz/file/j3ok3BAZ#rMD_5ekzuhM2phafzDrn3OUHq_ijACT_l58uES3UOR8). I've reverted the pack for now but I'm wondering if there's some mod conflict at play

EDIT: Nevermind, I was playing on the 0.1.1 version of ATM9 with the most recent Minecolonies version and I encountered the bug again after reloading my world following a freeze

**Comment by Raycoms** [comment]
Hmm, do you have a better log than this new one? It doesnt show the thread dumb this time, nor any exception unfortunately

**Comment by NoMoreStars** [comment]
That's about this best I can give you I'm afraid since it isn't even a proper crash, the world just never loads. Might just abandon it at this point.

**Comment by Raycoms** [comment]
If you wait for around 60s, the world should die on its own and give us a stracktrace

**Comment by someaddons** [comment]
holding F3+c for 10seconds should generate a manual crash report

**Comment by NoMoreStars** [comment]
[Here](https://mclo.gs/7ParcVb) we go! Had to update to the latest version of the pack including minecolonies since it wasn't outputting the crashlog otherwise for some reason

**Comment by Raycoms** [comment]
Odd, this doesnt mention us at all in the crash report

**Comment by RianAshes** [comment]
Hi! I have the same issue. Disabling minecolonies loads world correctly. Disabling modern fix does nothnig.
Can't generate crashlog, f3+c not responding, so there is [latest.log](https://github.com/ldtteam/minecolonies/files/13446428/latest.log) and some stacktrace inside it. 
If you need some additional info - let me know, i'll do my best.

Hope you can fix it soon. 

There is minecolonies-related part of log

```
"Server thread" prio=4 Id=274 WAITING on java.lang.Object@40792f1c
	at java.base@17.0.8/java.lang.Object.wait(Native Method)
	-  waiting on java.lang.Object@40792f1c
	at java.base@17.0.8/java.lang.Object.wait(Object.java:338)
	at TRANSFORMER/structurize@1.20.1-1.0.662-BETA/com.ldtteam.structurize.util.ManualBarrier.waitOne(ManualBarrier.java:31)
	at TRANSFORMER/structurize@1.20.1-1.0.662-BETA/com.ldtteam.structurize.storage.StructurePacks.waitUntilFinishedLoading(StructurePacks.java:72)
	at TRANSFORMER/structurize@1.20.1-1.0.662-BETA/com.ldtteam.structurize.storage.StructurePacks.getBlueprint(StructurePacks.java:392)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingMiner.needsFallBack(BuildingMiner.java:351)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingMiner.initStructure(BuildingMiner.java:294)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.miner.EntityAIStructureMiner.executeStructurePlacement(EntityAIStructureMiner.java:763)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.miner.EntityAIStructureMiner.searchANodeToMine(EntityAIStructureMiner.java:728)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.miner.EntityAIStructureMiner.executeNodeMining(EntityAIStructureMiner.java:650)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.miner.EntityAIStructureMiner$$Lambda$46393/0x0000000804bb77a8.get(Unknown Source)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicTransition.getNextState(BasicTransition.java:79)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicStateMachine.transitionToNext(BasicStateMachine.java:163)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicStateMachine.checkTransition(BasicStateMachine.java:149)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.checkTransition(TickRateStateMachine.java:133)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.tick(TickRateStateMachine.java:109)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.tick(AbstractAISkeleton.java:61)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.CitizenAI.lambda$registerWorkAI$4(CitizenAI.java:96)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.ai.citizen.CitizenAI$$Lambda$48057/0x0000000804d8ca88.get(Unknown Source)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicTransition.getNextState(BasicTransition.java:79)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicStateMachine.transitionToNext(BasicStateMachine.java:163)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicStateMachine.checkTransition(BasicStateMachine.java:149)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.checkTransition(TickRateStateMachine.java:133)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.tick(TickRateStateMachine.java:109)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.citizen.EntityCitizen.lambda$new$11(EntityCitizen.java:299)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.citizen.EntityCitizen$$Lambda$46345/0x0000000804badec0.getAsBoolean(Unknown Source)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicTransition.checkCondition(BasicTransition.java:87)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.basestatemachine.BasicStateMachine.checkTransition(BasicStateMachine.java:138)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.checkTransition(TickRateStateMachine.java:133)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.ai.statemachine.tickratestatemachine.TickRateStateMachine.tick(TickRateStateMachine.java:109)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.citizen.AbstractEntityCitizen.m_8107_(AbstractEntityCitizen.java:419)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.coremod.entity.citizen.EntityCitizen.m_8107_(EntityCitizen.java:725)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2292)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.m_8119_(Mob.java:337)
	at TRANSFORMER/minecolonies@1.20.1-1.1.321-BETA/com.minecolonies.api.entity.citizen.AbstractCivilianEntity.m_8119_(AbstractCivilianEntity.java:88)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:693)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel$$Lambda$48028/0x0000000804d7a408.accept(Unknown Source)
	at TRANSFORMER/noseenotick@2.0.0-build.9999/net.minecraft.entity.TickOptimizer.handleGuardEntityTick(TickOptimizer.java:62)
	at TRANSFORMER/noseenotick@2.0.0-build.9999/net.minecraft.entity.TickOptimizer.entityTicking(TickOptimizer.java:25)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.Level.m_46653_(Level.java:5541)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:343)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel$$Lambda$47904/0x0000000804d4c000.accept(Unknown Source)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:323)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:893)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:89)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer$$Lambda$45265/0x0000000804a161a0.run(Unknown Source)
	at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)
```

**Comment by RianAshes** [comment]
found a workaround.
1. Backup world
2. Remove 'entities' folder, disable minecolonies mod.
3. Load world. Exit world.
4. Enable minecolonies mod. Restart game.
5. Restore 'region' folder from backup.
6. Load world. Exit world.
7. Restore 'entities' folder from backup.
8. Load world.
9. ???
10. Profit.

After that world can be loaded correctly, no colonist or colony buildings has harmed. I guess that it's some kind of AI issue, and this operations clears AI state of colonists.

**Comment by Raycoms** [comment]
AI is reset on load, it must be some corruption of a minecolonies entity in a vanilla way or so, no clue

**Comment by Egos90** [comment]
Problem still exist on ATM 9 0.2.25 pack. With latest version of minecolonies-1.20.1-1.1.343-BETA
Some logs:

Latest:
https://gist.github.com/Egos90/9e67515b08bf339f3696a5ec9c108ccf

Without Modernfix:
https://gist.github.com/Egos90/21c1532b0ad26057d8060936cd3eba55 Without modernfix still not loading world

**Comment by MotionlessTrain** [comment]
This has been fixed in 1.20 after that version

It looks like Curseforge is slow, so 1.0.344 doesn't seem to be listed yet

**Comment by Egos90** [comment]
Thankyou for realy fast answer 👍 So I'll wait

Or trying [RianAshes](https://github.com/RianAshes) Workaround
